supervisor/hyperpod.go

```diff
@@ -408,6 +409,20 @@ func (hp *HyperPod) createContainer(container, bundlePath, stdin, stdout, stderr
 
 	glog.Infof("createContainer() calls c.run(p)")
 	c.run(p)
+	alarm := time.After(5 * time.Second)
+	for {
+		if pid := c.ownerPod.getNsPid(); pid > 0 {
+			p.ProcId = pid
+			break
+		}
+
+		select {
+		case <-alarm:
+			glog.V(3).Infof("Timeout while waiting for container's pid")
+		default:
+			time.Sleep(100 * time.Millisecond)
+		}
+	}
```

Comment:
`sv.CreateContainer() -> sv.getHyperPod() -> createHyperPod() -> hp.startNsListener() -> cmd.Start()` all are directly calls, after sv.CreateContainer() returns, the proxy/pid was created.
`hp.createContainer()` is after `sv.getHyperPod()` so the code in `hp.createContainer()` can sees the proxy/pid.